src/GraphQL.MemoryCache/AutomaticPersistedQueries/AutomaticPersistedQueriesExecutionBase.cs

```diff
@@ -0,0 +1,135 @@
+public abstract class AutomaticPersistedQueriesExecutionBase : IConfigureExecution
+{
+    /// <summary>
+    /// Supported version.
+    /// </summary>
+    public const string SUPPORTED_VERSION = "1";
+
+    private SHA256? _sha256;
+
+    /// <summary>
+    /// Searching APQ fields in <see cref="ExecutionOptions.Extensions"/> based on a protocol:
+    /// https://www.apollographql.com/docs/react/api/link/persisted-queries/#protocol
+    /// </summary>
+    public virtual (string? Hash, string? Version, bool Enabled) GetAPQProperties(Inputs? extensions)
+    {
+        string? hashResult = null;
+        string? versionResult = null;
+        bool enabledResult = false;
+
+        if (extensions?.TryGetValue("persistedQuery", out var persistedQueryObject) ?? false)
+        {
+            enabledResult = true;
+
+            if (persistedQueryObject is Dictionary<string, object> persistedQuery)
+            {
+                if (persistedQuery.TryGetValue("sha256Hash", out var sha256HashObject) && sha256HashObject is string sha256Hash && !string.IsNullOrEmpty(sha256Hash))
+                {
+                    hashResult = sha256Hash;
+                }
+
+                if (persistedQuery.TryGetValue("version", out var versionObject) && versionObject is string version && !string.IsNullOrEmpty(version))
+                {
+                    versionResult = version;
+                }
+            }
+        }
+
+        return (hashResult, versionResult, enabledResult);
+    }
+
+    /// <summary>
+    /// Check equality of the provided hash and a hash computed from the query.
+    /// </summary>
+    public virtual bool CheckHash(string hash, string query)
+    {
+        var inputBytes = Encoding.UTF8.GetBytes(query);
+        var shaShared = Interlocked.Exchange(ref _sha256, null) ?? SHA256.Create();
+        var bytes = shaShared.ComputeHash(inputBytes);
+        Interlocked.CompareExchange(ref _sha256, shaShared, null);
+
+#if NET5_0_OR_GREATER
+        var queryHash = Convert.ToHexString(bytes);
+#else
+        var builder = new StringBuilder(bytes.Length * 2);
+        foreach (var item in bytes)
+        {
+            builder.Append(item.ToString("x2"));
+        }
+        var queryHash = builder.ToString();
+#endif
+
+        return hash.Equals(queryHash, StringComparison.InvariantCultureIgnoreCase);
+    }
+
+    /// <summary>
+    /// Create <see cref="ExecutionResult"/> with provided error.
+    /// </summary>
+    public virtual ExecutionResult CreateExecutionResult(ExecutionError error) => new ExecutionResult { Errors = new ExecutionErrors { error } };
+
+    /// <summary>
+    /// Get query by hash. It is likely to be implemented via cache mechanism.
+    /// </summary>
+    public abstract ValueTask<string?> GetQueryAsync(string hash);
+
+    /// <summary>
+    /// Set query by hash. It is likely to be implemented via cache mechanism.
+    /// </summary>
+    public abstract Task SetQueryAsync(string hash, string query);
```

Comment:
Change to `protected`?

Reply:
@Shane32 What do you think? I would change all methods to protected, merge and continue with PR from server project.

Reply:
fix: merge and publish v5.3.0

Reply:
@sungam3r I agree the methods should be protected

Reply:
@SlavaUtesinov Let's do this final change and I'm glad to merge this awesome feature.

Reply:
done